### DISCUSSION TITLE
New in pre-release: commands with arguments

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#4394] [Announcements]
By creating [custom keyboard shortcuts](https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts), you can bind a shortcut to a [command](https://www.sumatrapdfreader.org/docs/Commands).

Until now the commands couldn't take arguments but it makes sense in some cases so since **ver 3.6** some commands can take arguments that affect their behavior.

One example: `CmdCreateAnnotHighlight` now takes a color argument so you can now change the highlight color by re-binding key `a` to e.g. command `CmdCreateAnnotHighlight #00ff00` (`#00ff00` is green color) or create multiple shortcuts with different colors.

See [commands with arguments](https://www.sumatrapdfreader.org/docs/Commands#commands-with-arguments) for a list of commands that take arguments and their arguments.

Let me know if you think other commands should also take arguments.

**Comment by H5820121** [comment]
@Kjk
"Let me know if you think other commands should also take arguments"
Yes,
1. The most important (for me) to change to ZOOM X, for example (who I use
a lot for my certain documents) ZOOM 175, now I have to press CTR+Y every
time and then write 175 and press Enter, it would be much more convenient
if everything was done with one shortcut key.

2. to give a shortcut key to change the scheme of TextColor &
BackgroundColor, for example those who want at night to read the
document  in dark mode (the background of the document is a shade of black
and the text color a shade of white) can easily switch the color scheme,
and during the day by a shortcut key return to a day color scheme.
Thanks!

‫בתאריך יום ג׳, 23 ביולי 2024 ב-20:20 מאת ‪Krzysztof Kowalczyk‬‏ <‪
***@***.***‬‏>:‬

**Reply by GitHubRulesOK** [reply]
@H5820121 you can zoom current page or current document (page is your need)
set one key to run something like  
```
SumatraPDF.exe -reuse-instance  -page %p -zoom 175 "%1"
```

this works for me in pdf
```
ExternalViewers [
	[
		CommandLine = "C:\program files\sumatrapdf\SumatraPDF.exe" -reuse-instance -page %p -zoom 175 "%1"
		Name = CMD
		Filter = *.pdf
		Key = z
	]
]
```
result after press z key
![image](https://github.com/user-attachments/assets/fd0e2e34-aec1-4a69-93c8-6a4972422580)



the foreground background toggle has always been simply invert which is back to the `i` key
![image](https://github.com/user-attachments/assets/4a16dc7f-f92e-4e4d-b187-51d5e14b11cb)

**Reply by H5820121** [reply]
@GitHubRulesOK 
This is a wonderful idea!, you mean through the ExternalViewers. The new option in the ExternalViewers setting is so powerful that I still haven't gotten used to it.

**Comment by H5820121** [comment]
@GitHubRulesOK 
"the foreground background toggle has always been simply invert which is back to the i key"
1. Where can I change the keyboard shortcut to this because I use a Hebrew keyboard?
2. Is it possible to change the shade of black and the shade of white?

**Reply by GitHubRulesOK** [reply]
> @GitHubRulesOK "the foreground background toggle has always been simply invert which is back to the i key"
> 
> 1. Where can I change the keyboard shortcut to this because I use a Hebrew keyboard?
> 2. Is it possible to change the shade of black and the shade of white?

1) I dont know which key is = `I` in Hebrew layout I have problems if I switch any to Arabic let alone Hebrew without any knowledge.

2) the traditional colour CLI call for background and foreground thus as many as you wish (end of list under depreciated) also worked last time I tried but that was when it was 3 keys and easier to call a batch file (years ago)

**Reply by GitHubRulesOK** [reply]
Hmmm I have problems with some commands as disadvantaged running 32 bit so not always sure why I can not run some settings in a live session they must be run with a fresh started setting. This was apparently due to a "fix" in the 32bit build process.

so switching to watchman yellow is fickle I suggest you DO NOT use this method it was depreciated long ago and I think more recently less useful with theming
```
SumatraPDF.exe" -set-color-range #ff0000 #ffff00 ..\document92.pdf
```
![image](https://github.com/user-attachments/assets/46fc3ec7-1abe-4507-8681-158c24ea3c9a)![image](https://github.com/user-attachments/assets/50ccd0c5-ea2c-49eb-b003-93b6ddee2da1)

**Comment by kjk** [comment]
@H5820121 see https://github.com/sumatrapdfreader/sumatrapdf/issues/4396 for zoom

**Comment by H5820121** [comment]
@kjk
I tried in V3.6.16461 64-bit:
[
Cmd = CmdZoomCustom 175 %
Key = 4
]
And it doesn't work properly, it just works as if CTR+Y were pressed.
Thanks!

‫בתאריך יום ד׳, 24 ביולי 2024 ב-17:05 מאת ‪Krzysztof Kowalczyk‬‏ <‪
***@***.***‬‏>:‬

**Reply by GitHubRulesOK** [reply]
@H5820121 you are too quick it will be tomorrow in that 16465+ (WIP) or more likely any later daily than current run 16469+

**Comment by H5820121** [comment]
@GitHubRulesOK <https://github.com/GitHubRulesOK>  Thanks!,
"SumatraPDF.exe" -set-color-range #dddddd #282828 -page %p "%1"" it no so
bad, you have only to use a new instance.


‫בתאריך יום ד׳, 24 ביולי 2024 ב-21:42 מאת ‪GitHubRulesOK‬‏ <‪
***@***.***‬‏>:‬
